common4j/src/main/com/microsoft/identity/common/java/platform/Device.java

```diff
@@ -101,8 +102,8 @@ public static String getProductVersion() {
 
         final String version = DiagnosticContext.INSTANCE.getRequestContext().get(AuthenticationConstants.SdkPlatformFields.VERSION);
         if (StringUtil.isNullOrEmpty(version)) {
-            Logger.warn(TAG + methodName, "Product version is not set.");
-            return NOT_SET;
+            Logger.warn(TAG + methodName, "Product version is not set.", null);
+            return StringUtil.isNullOrEmpty(BuildConfig.VERSION_NAME) ? "1.5.9-default" : BuildConfig.VERSION_NAME + "-default";
```

Comment:
Why are we hardcoding 1.5.9-default here?

NOT_SET means we're not initializing this properly. This version value is also platform-dependent. Not sure where 1.5.9 comes from.

Reply:
This was changed in #1607

Reply:
Let's add a comment + create an PBI to fix this properly.

Reply:
to 
1. undo the default value to NOT_SET (if we're getting this value, it means the upper layer didn't initialize the value properly).
2. Fix it in the upper layer(s).